frontend/src/metabase/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,29 @@
+import React from "react";
+import { RoutelessFullPageModal } from "metabase/components/Modal/RoutelessFullPageModal";
+import {
+  WindowModal,
+  WindowModalProps,
+} from "metabase/components/Modal/WindowModal";
+import { FullPageModalProps } from "metabase/components/Modal/FullPageModal";
+
+const Modal = ({
+  full = false,
+  ...props
+}: {
+  full?: boolean;
+  isOpen?: boolean;
+} & (WindowModalProps & FullPageModalProps)) =>
+  full ? (
+    props.isOpen ? (
+      <RoutelessFullPageModal {...props} />
+    ) : null
+  ) : (
+    <WindowModal {...props} />
+  );
```

Comment:
some if statements would probably be a bit easier to read than this nested ternary

Reply:
Absolutely agree 😄